Quick fix for zstreambuf

ice0 opened the pull request:
The problem was in `inflateInit2` incorrect initialization.
In this case we have no file header, quote from documentation:

> windowBits can also be –8..–15 for raw inflate. In this case, -windowBits
determines the window size. inflate() will then process raw deflate data,
not looking for a zlib or gzip header, not generating a check value, and
not looking for any check values for comparison at the end of the
stream. This is for use with other formats that use the deflate
compressed data format such as zip.

Anyway this change should be tested before the merge.

rodolforg commented:
Hm.. it crashes.
It does not allow to open Synfig `.sfg` anymore too.

rodolforg commented:
Please take a look: https://github.com/rodolforg/synfig/commit/edc6ece39955d7d35fe2685dd6c5055f869d670b

ice0 commented:
> Please take a look: [rodolforg@edc6ece](https://github.com/rodolforg/synfig/commit/edc6ece39955d7d35fe2685dd6c5055f869d670b)

Can't reproduce. Can you attach the problematic file?

rodolforg commented:
Unzip and try to import both files:
- One is a SFG (synfig history)
- Another is a Krita (a zip file, and it will try to load mergedimage.png)

[SynfigZipFiles.zip](https://github.com/synfig/synfig/files/8008223/SynfigZipFiles.zip)

ice0 commented:
@rodolforg 
Fixed in this commit: 0943ab14795a0ea908e72198f5e5d06377f4aead (Sometimes garbage data was left at the end of the buffer)

Also i changed `png_mptr::get_frame`
```
if (zip_fs && zipped_file.filename.empty()) {
```

rodolforg commented on synfig-core/src/synfig/zstreambuf.cpp:
```diff
@@ -158,35 +160,41 @@ size_t zstreambuf::unpack(void *dest, size_t dest_size, const void *src, size_t
+		// Note that Z_BUF_ERROR is not fatal, and deflate() can be called again
+		// with more input and more output space to continue compressing.
+		if (ret != Z_OK && ret != Z_STREAM_END && ret != Z_BUF_ERROR) {
+			//std::cerr << "Error: " << ret << ": " << inflate_stream_.msg << std::endl;
```
Delete this comment?

rodolforg commented:
Please rebase XD

rodolforg commented:
> Also i changed png_mptr::get_frame

You didn't, didn't you?
As far as I can see, it still works.
